src/main/scala/cromwell/engine/package.scala

```diff
@@ -34,4 +34,9 @@ package object engine {
     override def toString: String = "Succeeded"
     override val isTerminal = true
   }
+
+  trait WorkflowDescriptor {
+    def name: String
+    def id: WorkflowId
+  }
```

Comment:
It absolutely is semantics. I just wanted a clear definition of what it's supposed to be before deciding what that should then encompass. 

I'm in favor of not using the trait approach, see my comments elsewhere about case classes.